libs/ktem/ktem/index/file/base.py

```diff
@@ -74,7 +74,9 @@ def run(
 
     def stream(
         self, file_paths: str | Path | list[str | Path], *args, **kwargs
-    ) -> Generator[Document, None, tuple[list[str | None], list[str | None]]]:
+    ) -> Generator[
+        Document, None, tuple[list[str | None], list[str | None], list[Document]]
```

Comment:
about the complexity that this function return, some of the internal type args should be modelled for better readability